app/core/api.js

```diff
@@ -470,17 +470,60 @@ function setPatientSettings(person, cb) {
 function updatePatient(patient, cb) {
   var patientId = patient.userid;
 
-  var profile = patient.profile;
-  tidepool.addOrUpdateProfile(patientId, profile, function(err, profile) {
-    if (err) {
-      return cb(err);
-    }
+  const profile = profileFromUser(patient);
+  const loggedInUserId = tidepool.getUserId();
+  const updatedEmails = _.get(patient, 'emails');
+  const newEmail = _.get(updatedEmails, '[0]');
+
+  if(loggedInUserId !== patientId){
+    tidepool.findProfile(patientId, function(err, currentProfile){
+      if(err){
+        return cb(err);
+      }
+      const currentEmail = _.get(currentProfile, 'emails[0]');
+      const isEmailUpdated = newEmail !== currentEmail;
+      if(isEmailUpdated){
+        return async.series([
+          function(callback){
+            tidepool.updateCustodialUser({username: newEmail, emails: updatedEmails}, patientId, callback);
+          },
+          function(callback){
+            tidepool.addOrUpdateProfile(patientId, profile, callback);
+          },
+          function(callback){
+            tidepool.signupStart(patient.userid, callback);
+          }
+        ], function(err, results){
+          if(err){
+            return cb(err);
+          }
+          const updatedPatient = userFromAccountAndProfile({ account: results[0], profile: results[1] });
+
+          return cb(null, updatedPatient)
+        })
+      } else {
```

Comment:
While it is essentially the same `else` block, this one relies on information from an additional API call. I could think of a way to DRY it out, but it would involve (as far as I can logic out) making the API call for every update as opposed to only updates to patients that are not the current user. I sort of figured that fewer API calls was worth the lack of DRYness. However, if you see an obvious way to go about it that I'm missing, I'm totally open to it.